clip_retrieval/clip_inference/distributor.py

```diff
@@ -2,27 +2,36 @@
 
 import os
 
+from .worker import worker
+
 
 class SequentialDistributor:
-    def __init__(self, runner, output_partition_count):
-        self.runner = runner
-        self.output_partition_count = output_partition_count
+    def __init__(self, worker_args):
```

Comment:
I understand the intent but generic args make things hard to understand

I think it may be better to do something like
- tasks
- worker_args

Reply:
Also tests will need to be adapted. That may help fixing things

Reply:
> I understand the intent but generic args make things hard to understand
> 
> I think it may be better to do something like
> 
> * tasks
> * worker_args

sure, so the tasks will be pre-computed in `clip_inference.main` and passed as a separate argument to each of the distributors?

furthermore...do you want to ditch the following in `clip_inference.main`?
```python
...
local_args = locals()
local_args.pop("wds_number_file_per_input_file")
local_args.pop("write_batch_size")
local_args.pop("distribution_strategy")
...
```

I thought it was a cute/clever way to grab all the arguments and pass them along--but I am not married to it and am okay with passing everything verbosely to each of the distributors, lmk what you think is better here

Reply:
here's the separate task argument https://github.com/rom1504/clip-retrieval/pull/184/commits/31936486a8b221f226c36a528cced030e0016593

Reply:
I think it's good to pass most of the argument as a dict/kwargs and the distribution/task ones explicitly